native/src/main/java/io/ballerina/lib/data/csvdata/csv/CsvParser.java

```diff
@@ -334,6 +391,181 @@ public Object execute(Reader reader, Type type, CsvConfig config, BTypedesc bTyp
+                printErrorLogs(environment, exception, keyValues);
+            }
+            printErrorLogs(environment, exception);
+        }
+
+        private static void handleLogFileGeneration(String filePath) {
+            try {
+                Path path = Paths.get(filePath);
+                Path parentDir = path.getParent();
+                if (parentDir != null) {
+                    Files.createDirectories(parentDir);
+                }
+                if (Files.notExists(path)) {
+                    Files.createFile(path);
+                }
+            } catch (IOException ioException) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.FAILED_FILE_IO_OPERATION,
+                        String.format(FILE_IO_ERROR, filePath, ioException.getMessage()));
+            }
+        }
+
+        private void printErrorLogs(Environment environment, Exception exception) {
+            StrandMetadata strandMetadata = new StrandMetadata(true,
+                    ModuleUtils.getProperties(PRINT_ERROR));
+            String errorMessage = String.format(CSV_PARSE_ERROR, this.lineNumber + 1,
+                    this.columnIndex + 1, exception.getMessage());
+            Object[] arguments = new Object[]{StringUtils.fromString(errorMessage),
+                    null, null, ValueCreator.createMapValue()};
+            environment.getRuntime().callFunction(ModuleUtils.getModule(), PRINT_ERROR,
+                    strandMetadata, arguments);
+        }
+
+        private void printErrorLogs(Environment environment, Exception exception, BMap<BString, Object> keyValues) {
+            StrandMetadata strandMetadata = new StrandMetadata(true,
+                    ModuleUtils.getProperties(PRINT_ERROR));
+            String errorMessage = String.format(CSV_PARSE_ERROR, this.lineNumber + 1,
+                    this.columnIndex + 1, exception.getMessage());
+            Object[] arguments = new Object[]{StringUtils.fromString(errorMessage),
+                    null, null, keyValues};
+            environment.getRuntime().callFunction(ModuleUtils.getModule(), PRINT_ERROR,
+                    strandMetadata, arguments);
+        }
+
+        private void overwriteLogFile(String filePath) {
+            Path path = Paths.get(filePath);
+            try {
+                Files.write(path, new byte[0], StandardOpenOption.CREATE, StandardOpenOption.TRUNCATE_EXISTING);
+            } catch (IOException exception) {
+                throw DiagnosticLog.error(DiagnosticErrorCode.FAILED_FILE_IO_OPERATION,
+                        String.format(FILE_OVERWRITE_ERROR, filePath, exception.getMessage()));
+            }
+        }
+
+        private void writeLogsToFile(String filePath, String content) {
+            Path path = Paths.get(filePath);
+            try {
+                Files.writeString(path, content.trim() + "\n", StandardOpenOption.CREATE,
```

Comment:
Should avoid platform-dependent characters - `\n` => This only works for linux. Use the lineSeparator API

Reply:
Fixed by [ba54cb5](https://github.com/ballerina-platform/module-ballerina-data.csv/pull/60/commits/ba54cb514a2ee33ed48a4e5c3251c549a82cb120)